src/main/java/org/skriptlang/skript/scheduler/AsyncTask.java

```diff
@@ -0,0 +1,185 @@
+	 */
+	public Plugin getPlugin() {
+		return plugin;
+	}
+
+	/**
+	 * @return The period of this task if it's repeating, otherwise -1
+	 */
+	public long getPeriod() {
+		return period;
+	}
+
+	/**
+	 * @return The amount the delay and period represent in TimeUnit.
+	 */
+	public TimeUnit getTimeUnit() {
+		return unit;
+	}
+
+	/**
+	 * @return Whether this task is still running, i.e. whether it will run later or is currently running.
+	 */
+	public final boolean isAlive() {
+		return TaskManager.isAlive(this);
+	}
+
+	/**
+	 * Cancels this task.
+	 * @return true if the task was successfully cancelled.
+	 */
+	public final boolean cancel() {
+		return TaskManager.cancel(this);
+	}
+
+	@Override
+	public void close() {
+		cancel();
+	}
+
+	/**
+	 * Re-schedules the task to run next after the given delay. If this task was repeating it will continue so using the same period as before.
+	 * 
+	 * @param delay The delay using the existing TimeUnit.
+	 */
+	public void setNextExecution(long delay) {
+		setNextExecution(delay, unit);
+	}
+
+	/**
+	 * Re-schedules the task to run next after the given delay. If this task was repeating it will continue so using the same period as before.
+	 * 
+	 * @param delay The delay using the existing TimeUnit.
+	 * @param unit The new TimeUnit you want to delay the new task by.
+	 */
+	public void setNextExecution(long delay, TimeUnit unit) {
+		assert delay >= 0;
+		cancel();
+		TaskManager.run(this, delay, unit);
+	}
+
+	/**
+	 * Sets the period of this task. This will re-schedule the task to be run next after the given period if the task is still running.
+	 * 
+	 * @param period Period in ticks or -1 to cancel the task and make it non-repeating.
+	 */
+	public void setPeriod(long period) {
+		setPeriod(period, unit);
+	}
+
+	/**
+	 * Sets the period of this task. This will re-schedule the task to be run next after the given period if the task is still running.
+	 * 
+	 * @param period Period in ticks or -1 to cancel the task and make it non-repeating.
+	 * @param unit The new TimeUnit you want to delay the new task by.
+	 */
+	public void setPeriod(long period, TimeUnit unit) {
+		assert period == -1 || period > 0;
+		if (period == this.period)
+			return;
+		this.period = period;
+		if (isAlive()) {
+			cancel();
+			if (period != -1)
+				TaskManager.run(this, period, unit);
+		}
+	}
+
+	/**
+	 * Calls a one off task that executes once with no delay.
+	 */
+	public static void run(Runnable runnable) {
+		new AsyncTask(0, TimeUnit.MILLISECONDS) {
+			@Override
+			public void run() {
+				runnable.run();
+			}
+		};
+	}
+
+	/**
+	 * Utility method for converting the provided milliseconds to ticks to work with the Spigot API.
+	 * 
+	 * @param milliseconds The amount of milliseconds to convert to ticks.
+	 * @return The amount of ticks the provided milliseconds argument contained.
+	 */
+	public static long millisecondsToTicks(long milliseconds) {
+		return (milliseconds / 1000) * 20;
```

Comment:
Why not / 50?

Reply:
Because Minecraft tick rate is 20?

Reply:
Personally I find it more confusing to go to seconds then ticks, rather than straight to ticks (plus it's an extra operation!)
but it's a minor thing so keep it as is if you want.

Reply:
It should probably use the new Timespan and Timespan#TimePeriod now that it exists.